@HOOptions/HOOptions.m

```diff
@@ -0,0 +1,322 @@
+            if islogical(value)
+                this.save = value;
+            else
+                error('HOOptions.save must be a logical value.');
+            end
+        end
+        
+        function set.n_exp(this, value)
+            if isscalar(value)
+                this.n_exp = value;
+                %warning('expgroups_noise and expgroups_scaling set to default');
+                this.expgroups_noise = {[1:this.n_exp]};
+                this.expgroups_scaling = {[1:this.n_exp]};
+            else
+                error('HOOptions.n_exp must be a scalar.');
+            end
+        end
+        
+        function set.n_obs(this, value)
+            if isscalar(value)
+                this.n_obs = value;
+                %warning('obsgroups_noise, obsgroups_scaling, and scale set to default');
+                this.obsgroups_noise = {[1:this.n_obs]};
+                this.obsgroups_scaling = {[1:this.n_obs]};
+                
+                this.scale = strings(1,this.n_obs);
+                this.scaling = strings(1,this.n_obs);
+                this.noise = strings(1,this.n_obs);
+                for i_obs = 1:this.n_obs
+                    this.scale{i_obs} = 'lin';
+                    this.scaling{i_obs} = 'single';
+                    this.noise{i_obs} = 'single';
+                end
+            else
+                error('HOOptions.n_obs must be a scalar.');
+            end
+        end
+        
+        function set.expgroups_noise(this, value)
+            if iscell(value)
+                this.expgroups_noise = value;
+            else
+                error('HOOptions.expgroups_noise must be a cell array.');
+            end
+        end
+        
+        function set.expgroups_scaling(this, value)
+            if iscell(value)
+                this.expgroups_scaling = value;
+            else
+                error('HOOptions.expgroups_scaling must be a cell array.');
+            end
+        end
+        
+        function set.obsgroups_scaling(this, value)
+            if iscell(value)
+                this.obsgroups_scaling = value;
+            else
+                error('HOOptions.obsgroups_scaling must be a cell array.');
+            end
+        end
+        
+        function set.obsgroups_noise(this, value)
+            if iscell(value)
+                this.obsgroups_noise = value;
+            else
+                error('HOOptions.obsgroups_noise must be a cell array.');
+            end
+        end
+        
+        function set.distribution(this, value)
+            if (strcmp(value, 'normal') || strcmp(value, 'laplace'))
+                this.distribution = value;
+            else
+                error('HOOptions.distribution must be set to either "normal" or "laplace".');
+            end
+        end
+        
+        function set.scale(this, value)
+            if ~isequal(this.n_obs,length(value))
+                error('HOOptions.scale must have the dimension 1xn_obs');
```

Comment:
~dimension 1xn_obs~ length n_obs

Reply:
also length(x) == max(size(X)), so the check you are doing here may fail to catch a problem if value is a matrix.